src/bundle/Resources/public/js/scripts/admin.location.visibility.js

```diff
@@ -1,12 +1,42 @@
-(function (global, doc) {
-    const updateVisibility = event => {
-        event.target.closest('.ez-checkbox-icon').classList.toggle('is-checked');
+(function(global, doc) {
+    const SELECTOR_VISIBILITY_CHECKBOXES = '#ez-tab-location-view-locations .ez-checkbox-icon__checkbox';
+    const SELECTOR_VISIBILITY_FORM = 'form[name="location_update_visibility_data"]';
+    const form = doc.querySelector(SELECTOR_VISIBILITY_FORM);
+    const visibilityCheckboxes = doc.querySelectorAll(SELECTOR_VISIBILITY_CHECKBOXES);
+    const onVisibilityUpdated = ({ target }) => {
+        const { checked: isVisible } = target;
 
+        target.closest('.ez-checkbox-icon').classList.toggle('is-checked', isVisible);
+    };
+    const handleUpdateError = global.eZ.helpers.notification.showErrorNotification;
+    const handleUpdateSuccess = ({ message }) => {
+        global.eZ.helpers.notification.showSuccessNotification(message);
+    };
+    const handleUpdateResponse = (event, response) => {
+        if (response.status !== 200) {
+            throw new Error(response.statusText);
+        }
+
+        onVisibilityUpdated(event);
```

Comment:
If you handle response here, then you should call this method in the next `then()`